service/src/main/java/bio/terra/tanagra/service/artifact/FeatureSetService.java

```diff
@@ -103,4 +103,27 @@ public FeatureSet updateFeatureSet(
         featureSetId, userEmail, displayName, description, criteria, outputAttributesPerEntity);
     return featureSetDao.getFeatureSet(featureSetId);
   }
+
+  public FeatureSet cloneFeatureSet(
+      String studyId,
+      String featureSetId,
+      String userEmail,
+      @Nullable String displayName,
+      @Nullable String description) {
+    FeatureSet original = getFeatureSet(studyId, featureSetId);
+    FeatureSet.Builder featureSetBuilder =
+        FeatureSet.builder()
+            .underlay(original.getUnderlay())
+            .displayName(displayName != null ? displayName : original.getDisplayName())
+            .description(description != null ? description : original.getDescription())
+            .createdBy(userEmail)
+            .lastModifiedBy(userEmail)
+            // Make a deep copy of Criteria since it has an id
+            .criteria(original.getCriteria().stream().map(Criteria::deepCopy).toList())
```

Comment:
OK to clarify, I wasn't trying to say you should change the id. I was trying to suggest you remove the deepCopy call. is there a benefit to changing the criteria id? I think no because they just need to be unique per concept set. If that's the case, I'd prefer to just leave it for simpler code. This will be much more copying when you do the cohort clone, where there are nested objects.

Reply:
here is my understanding and subsequent reasoning - 
criteria is identified by its id and is stored in the DB as a row. It can be modified after creation. 
If the same id is used, when the cloned feature set is opened and its criteria is updated -- it will also reflect in the criteria of the original feature set. 
hence the approach to deep clone and create new criteria with new id

Reply:
tested it locally. Since there is also the conset_set_Id is in the table it does not over write changes with a shal copy. removing deep clone